src/test/java/org/opensearch/security/authtoken/jwt/JwtVendorTest.java

```diff
@@ -0,0 +1,120 @@
+        Settings settings = Settings.builder().put("signing_key", "abc123").build();
+
+        JsonWebKey jwk = JwtVendor.createJwkFromSettings(settings);
+        Assert.assertEquals("HS512", jwk.getAlgorithm());
+        Assert.assertEquals("sig", jwk.getPublicKeyUse().toString());
+        Assert.assertEquals("abc123", jwk.getProperty("k"));
+    }
+
+    @Test(expected = Exception.class)
+    public void testCreateJwkFromSettingsWithoutSigningKey() throws Exception {
+        Settings settings = Settings.builder().put("jwt", "").build();
+        JwtVendor.createJwkFromSettings(settings);
+    }
+
+    @Test
+    public void testCreateJwtWithRoles() throws Exception {
+        String issuer = "cluster_0";
+        String subject = "admin";
+        String audience = "audience_0";
+        List<String> roles = List.of("IT", "HR");
+        List<String> backendRoles = List.of("Sales");
+        String expectedRoles = "IT,HR";
+        Integer expirySeconds = 300;
+        LongSupplier currentTime = () -> (int) 100;
+        String claimsEncryptionKey = RandomStringUtils.randomAlphanumeric(16);
+        Settings settings = Settings.builder().put("signing_key", "abc123").put("encryption_key", claimsEncryptionKey).build();
+        Long expectedExp = currentTime.getAsLong() + expirySeconds;
+
+        JwtVendor jwtVendor = new JwtVendor(settings, Optional.of(currentTime));
+        String encodedJwt = jwtVendor.createJwt(issuer, subject, audience, expirySeconds, roles, backendRoles);
+
+        JwsJwtCompactConsumer jwtConsumer = new JwsJwtCompactConsumer(encodedJwt);
+        JwtToken jwt = jwtConsumer.getJwtToken();
+
+        Assert.assertEquals("cluster_0", jwt.getClaim("iss"));
+        Assert.assertEquals("admin", jwt.getClaim("sub"));
+        Assert.assertEquals("audience_0", jwt.getClaim("aud"));
+        Assert.assertNotNull(jwt.getClaim("iat"));
+        Assert.assertNotNull(jwt.getClaim("exp"));
+        Assert.assertEquals(expectedExp, jwt.getClaim("exp"));
+        Assert.assertNotEquals(expectedRoles, jwt.getClaim("er"));
+        Assert.assertEquals(expectedRoles, EncryptionDecryptionUtil.decrypt(claimsEncryptionKey, jwt.getClaim("er").toString()));
+    }
+
+    @Test(expected = Exception.class)
+    public void testCreateJwtWithBadExpiry() throws Exception {
+        String issuer = "cluster_0";
+        String subject = "admin";
+        String audience = "audience_0";
+        List<String> roles = List.of("admin");
+        Integer expirySeconds = -300;
+        String claimsEncryptionKey = RandomStringUtils.randomAlphanumeric(16);
+
+        Settings settings = Settings.builder().put("signing_key", "abc123").put("encryption_key", claimsEncryptionKey).build();
+        JwtVendor jwtVendor = new JwtVendor(settings, Optional.empty());
+
+        jwtVendor.createJwt(issuer, subject, audience, expirySeconds, roles, List.of());
+    }
+
+    @Test(expected = Exception.class)
```

Comment:
HI, @RyanL1997, Could you plz use `assertThrows()` Junit 4 has it as well?. In this case it is a false-positive test, since the `Exception.class` is too broad or you need to use the different exception. If it it fails  in line #99 or #95 it will give the wrong results.

Reply:
Hi @willyborankin, thanks for the feedback! Just for my knowledge isn't that the `assertThrows()` is coming from Junit5? However, I do have a version for fixing this, by manually implemented an `assertThrows` function. (Here is the [commit](https://github.com/opensearch-project/security/pull/3179/commits/17138dcfc1ecc5efcc1845a773149c748d899efa)). But then I was realized that I can use the `try/catch` with `assertEquals`.

Reply:
I use in my PR clean up PR:
```java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertThrows;
```
it works.

Reply:
For Junit 4 you can use @Rule.
```java
@Rule
public ExpectedException exceptionRule = ExpectedException.none();

@Test
public void whenExceptionThrown_thenRuleIsApplied() {
    exceptionRule.expect(NumberFormatException.class);
    exceptionRule.expectMessage("For input string");
    Integer.
```